arch/arc/core/reset.S

```diff
@@ -121,6 +121,36 @@ invalidate_dcache:
 
 done_cache_invalidate:
 
+/*
+ * ARC Custom Initialization
+ * For scenarios where board hardware is not reset between tests,
+ * Some settings need to be reset by software.
```

Comment:
I'd prefer to not use a word "reset" because we don't reset anything here in a typical sense.. i.e. HW is not being re-initialized internally. Instead we return its settings to the default initial state trying to substitute normal HW reset sequence.